src/net/rpc.rs

```diff
@@ -324,15 +324,15 @@ impl RPCPoxInfoData {
 
         let pox_consts = &burnchain.pox_constants;
 
-        if prepare_cycle_length != pox_consts.prepare_length as u64 {
+        if mainnet && prepare_cycle_length != pox_consts.prepare_length as u64 {
```

Comment:
While this would work insofar as stopping the bug from manifesting, I don't think this is correct.  Why is `pox_consts` not getting initialized to the same values as the result of calling `get-pox-info`?  I think they really need to be the same, since clients that use `/v2/info` to learn about the PoX configuration expect that this endpoint reflects how the `.pox` contract actually behaves.

Reply:
We are using the PoxConstants that are defined for regtest for the mocknet. These are defined in `src/burnchain/mod.rs`: 
```
pub fn regtest_default() -> PoxConstants {
        PoxConstants::new(
            5,
            1,
            1,
            3333333333333333,
            1,
            BITCOIN_REGTEST_FIRST_BLOCK_HEIGHT + POX_SUNSET_START,
            BITCOIN_REGTEST_FIRST_BLOCK_HEIGHT + POX_SUNSET_END,
        )
    }
```

I can change it so that the constants for regtest (and testnet?) match the values obtained from calling `get-pox-info`.

Reply:
It's the other way around -- you'll need to make sure that the `.pox` contract is initialized with these `PoxConstant` values.  Then this code path should work without modification.